src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -0,0 +1,325 @@
+    const index = this.children.indexOf(child);
+    if (index !== -1) {
+      this.children.splice(index, 1);
+    }
+    const beforeIndex = this.children.indexOf(beforeChild);
+    if (beforeIndex === -1) {
+      throw new Error('This child does not exist.');
+    }
+    this.children.splice(beforeIndex, 0, child);
+  }
+
+  update(type, props) {
+    this.type = type;
+    this.props = props;
+  }
+
+  toJSON() {
+    const {children, ...props} = this.props;
+    const json: ReactTestRendererJSON = {
+      type: this.type,
+      props: props,
+      children: null,
+    };
+
+    Object.defineProperty(json, '$$typeof', {value: Symbol.for('react.test.json')});
+    if (typeof children === 'string') {
+      json.children = [children];
+    } else {
+      var childrenJSON = [];
+      this.children.forEach((child) => {
+        if (typeof child.toJSON === 'function') {
+          childrenJSON.push(child.toJSON());
+        }
+      });
+      json.children = childrenJSON.length ? childrenJSON : null;
+    }
+    return json;
+  }
+}
+
+type Container = TestContainer;
+type Props = Object;
+type Instance = TestComponent;
+type TextInstance = {
+  text: string | number,
+  id: number,
+  rootContainerInstance: Container,
+  toJSON(): string | number,
+};
+
+var TestRenderer = ReactFiberReconciler({
+  getRootHostContext() {
+    return emptyObject;
+  },
+
+  getChildHostContext() {
+    return emptyObject;
+  },
+
+  prepareForCommit() : void {
+    // noop
+  },
+
+  resetAfterCommit() : void {
+    // noop
+  },
+
+  createInstance(
+    type : string,
+    props : Props,
+    rootContainerInstance : Container,
+    hostContext : Object,
+    internalInstanceHandle : Object,
+  ) : Instance {
+    return new TestComponent(type, props, rootContainerInstance);
+  },
+
+  appendInitialChild(parentInstance : Instance, child : Instance | TextInstance) : void {
+    parentInstance.appendChild(child);
+  },
+
+  finalizeInitialChildren(
+    testElement : Instance,
+    type : string,
+    props : Props,
+    rootContainerInstance : Container,
+  ) : boolean {
+    return false;
+  },
+
+  prepareUpdate(
+    testElement : Instance,
+    type : string,
+    oldProps : Props,
+    newProps : Props,
+    hostContext : Object,
+  ) : boolean {
+    return true;
+  },
+
+  commitUpdate(
+    instance : Instance,
+    type : string,
+    oldProps : Props,
+    newProps : Props,
+    rootContainerInstance : Container,
+    internalInstanceHandle : Object,
+  ) : void {
+    instance.update(type, newProps);
+  },
+
+  commitMount(
+    instance : Instance,
+    type : string,
+    newProps : Props,
+    rootContainerInstance : Object,
+    internalInstanceHandle : Object
+  ) : void {
+    // Noop
+  },
+
+  shouldSetTextContent(props : Props) : boolean {
+    return (
+      typeof props.children === 'string' ||
+      typeof props.children === 'number'
+    );
+  },
+
+  resetTextContent(testElement : Instance) : void {
+    // Noop
```

Comment:
Why is this a no-op? It seems like there could be bugs related to this.
(e.g. DOM renderer implements this one.)

I think we should actually have `shouldSetTextContent` always return `false`.
AFAIK it only exists as an optimization to avoid creating additional Fibers for single-text-child divs.
But in case of test renderer it doesn't matter.